crates/ark/src/viewer.rs

```diff
@@ -41,17 +51,51 @@ fn emit_html_output(iopub_tx: Sender<IOPubMessage>, path: String) -> Result<()>
 }
 
 #[harp::register]
-pub unsafe extern "C" fn ps_html_viewer(url: SEXP) -> anyhow::Result<SEXP> {
+pub unsafe extern "C" fn ps_html_viewer(
+    url: SEXP,
+    label: SEXP,
+    height: SEXP,
+    is_plot: SEXP,
+) -> anyhow::Result<SEXP> {
     // Convert url to a string; note that we are only passed URLs that
     // correspond to files in the temporary directory.
     let path = RObject::view(url).to::<String>();
+    let label = match RObject::view(label).to::<String>() {
+        Ok(label) => label,
+        Err(_) => String::from("R"),
+    };
     match path {
         Ok(path) => {
-            // Emit the HTML output
+            // Emit HTML output
             let main = RMain::get();
             let iopub_tx = main.get_iopub_tx().clone();
-            if let Err(err) = emit_html_output(iopub_tx, path) {
-                log::error!("Failed to emit HTML output: {:?}", err);
+            if main.session_mode == SessionMode::Notebook {
+                // In notebook mode, send the output as a Jupyter display_data message
+                if let Err(err) = emit_html_output_jupyter(iopub_tx, path, label) {
+                    log::error!("Failed to emit HTML output: {:?}", err);
+                }
```

Comment:
What "mode" is ark going to run in for Jupyter Console?

Reply:
A good question ... I think it defaults to console mode, and we should switch it to notebook mode (also possible to just add session mode to the `argv` when registering the kernel).